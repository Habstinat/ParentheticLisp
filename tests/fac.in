tests:
    - recursion with define
    - mult
    - shadowing built-ins

(
    define
    (() ()())

    fac
    (() ()()(()))

    (
        lambda
        (() ())

        n
        ( (() (())()) )

        (   if
            (() ()()())

            (   equal
                (() (())(()))
                n
                (() (())())
                1
                ((()) ())
            )

            1
            ((()) ())

            (   
                *
                (() ()(()))

                n
                (()(())())

                fac[n-1]
                (
                    (() ()()(()))
                    (
                        minus
                        (() (()()))

                        (()(())())

                        ((()) ())
                    )
                )
            )
        )
    )
)

(   
    fac
    (() ()()(()))

    10
    ((()) ()()()()()()()()()())

)

division [to test that div is unshadowed]
(
    (() (())())

    ((()) ()()()()()()()()())

    ((()) ()()()())
)