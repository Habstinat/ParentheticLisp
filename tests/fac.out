3628800.0
2.25